pkg/controller/node/nodecontroller.go

```diff
@@ -532,6 +532,32 @@ func (nc *NodeController) tryUpdateNodeStatus(node *api.Node) (time.Duration, ap
 				readyCondition.LastTransitionTime = nc.now()
 			}
 		}
+
+		// Like NodeReady condition, NodeOutOfDisk was last set longer ago than gracePeriod, so update
+		// it to Unknown (regardless of its current value) in the master, without contacting kubelet.
+		oodCondition := nc.getCondition(&node.Status, api.NodeOutOfDisk)
+		if oodCondition == nil {
+			glog.V(2).Infof("Out of disk condition of node %v is never updated by kubelet", node.Name)
+			oodCondition := &api.NodeCondition{
```

Comment:
There is no good reason now to do this in a different way. I tried to do something to make this logic clear, but abandoned that approach. This difference is a left over from what I was trying to do.

Reverting to make this look like readyCondition.